### PR TITLE
Wavlength calibration from raw data

### PR DESCRIPTION
**Pull request by YuiKasagi**
The file "examples/python/reduc_thar.py" is an example of the wavelength calibration part of the IRD data reduction. (To run this, some files such as raw fits files of ThAr are needed. Should I put them somewhere?) 
The "wavcal.py" of the previous PR started with a median combined ThAr file, but this example shows the almost whole process from the raw ThAr data to the wavelength calibrated ThAr spectrum. To allow wavelength calibration for arbitrary data, "wavcal.py" has been modified.

### CODE REVIEW

File: examples/python/reduc_thar.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from scipy.signal import medfilt
+import pathlib
+import numpy as np
+import pandas as pd
+import pkg_resources
+
+import argparse
+import matplotlib.pyplot as plt
+
+def imcorrect(date,im):
```
Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:24]: `date` is not used?

Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:24]: `date` is not used?

File: examples/python/reduc_thar.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    y0, interp_function, xmin, xmax, coeff = read_trace_file(pathC) #([pathC, path_c])
+    mask = trace(im, trace_legendre, y0, xmin, xmax, coeff)
+    calim[mask] = np.nan
+    calim[hotpix_mask] = np.nan
+    model_im = median_XY_profile(calim,show=plot)
+    corrected_im = im-model_im
+    #corrected_detector(im, model_im, corrected_im)
+    return corrected_im
+
+def im_to_rsd(date,corrected_im,mode='mmf2'):
```
Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:41]: `date` is not used?

Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:41]: `date` is not used?

File: examples/python/reduc_thar.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    parser.add_argument('--thar', type=int, nargs=2,default=[14893, 14942], help='thar_star frame; start, stop')
+    parser.add_argument('--step', type=int, nargs=1,default=[1],help='thar_star step of frame number')
+    parser.add_argument('--mode', nargs=1,default=['mmf2'], help='mmf1 (lfc fiber) or mmf2 (star fiber)')
+    parser.add_argument('--plot',default=False, help='plot')
+
+    args = parser.parse_args()
+    date = args.d[0]
+    fdark = args.dark
+    fthar = list(range(args.thar[0],args.thar[1],args.step[0]))
+    mode = arge.mode[0]
```
Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:72]: `args`

Comment (HajimeKawahara) [inline examples/python/reduc_thar.py:72]: `args`

**Review by HajimeKawahara — APPROVED** [review]
Confirmed that reduc_thar.py, pytests, and wavcal.py worked. It looks great!
Thanks.

**Comment by YuiKasagi** [comment]
Thank you for confirming! Comments are reflected.
In addition, the fitting function can be selected from 'chebyshev' or 'legendre' in wavcal.py.

**Pull request merged**